packages/containers/src/ComponentForm/kit/flatten.js

```diff
@@ -0,0 +1,50 @@
+/**
+ *  Copyright (C) 2006-2018 Talend Inc. - www.talend.com
+ *
+ *  Licensed under the Apache License, Version 2.0 (the 'License');
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an 'AS IS' BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+/**
+ * flatten an object means each keys are a jsonpath.
+ * jsperf: https://jsperf.com/talend-flatten
+ * @param {object} obj the source object
+ * @return {object} flatten object
+ * @example
+ * flatten({ level1: { level2: 'foo' }})
+ * // { 'level1.level2': 'foo' }
+ */
+export default function flatten(obj) {
```

Comment:
doc doesn't reflect that it is a deep flatten operation, taking care of array and objects